### ISSUE TITLE
Incorrect warning icons when ReferenceOutputAssembly="false"

### ISSUE THREAD

**Issue by jnm2** [dotnet/project-system#2928]
Whenever I use `ReferenceOutputAssembly="false"`, no matter what the target framework is (even between csproj and wixproj), I get a warning symbol even though there are no build warnings:

![image](https://user-images.githubusercontent.com/8040367/32134961-209b6010-bbc5-11e7-958e-e834632913d5.png)

In this instance:

#### ClassLibrary1.csproj:
```xml
<Project Sdk="Microsoft.NET.Sdk">

  <PropertyGroup>
    <TargetFramework>net462</TargetFramework>
  </PropertyGroup>

  <ItemGroup>
    <ProjectReference Include="..\ClassLibrary2\ClassLibrary2.csproj" ReferenceOutputAssembly="false" />
  </ItemGroup>

</Project>
```

#### ClassLibrary2.csproj:
```xml
<Project Sdk="Microsoft.NET.Sdk">

  <PropertyGroup>
    <TargetFramework>net462</TargetFramework>
  </PropertyGroup>

</Project>
```

**Comment by natidea** [comment]
DevNotes: with ReferenceOutputAssembly="false", NuGet does not generate a project reference in the assets file:
```json
      "frameworks": {
        "netstandard2.0": {
          "projectReferences": {}
        }
      },
```
This likely means a ResolvedProjectReference item is not produced. However this does not seem to be a problem in [Microsoft.NET.Build.Extensions.Tasks.csproj](https://github.com/dotnet/sdk/blob/4dd61b21e00a98c7a17111da934082be7bb4a667/src/Tasks/Microsoft.NET.Build.Extensions.Tasks/Microsoft.NET.Build.Extensions.Tasks.csproj#L39) which resolves its project reference correctly. Investigating further...

**Comment by evgeny-burmakov** [comment]
Do we have any workaround for hiding icon or reference?

**Comment by evgeny-burmakov** [comment]
Any news?

**Comment by drewnoakes** [comment]
@evgeny-burmakov thanks for your patience. It's definitely on the radar. This issue will be updated as soon as there is any progress.

**Comment by ReubenBond** [comment]
I believe this is the reason why we see warning icons in VS in [Orleans](https://github.com/dotnet/orleans). We use a `ProjectReference` with `SkipGetTargetFrameworkProperties` set in order to ensure correct build order (codegen must be built before projects which rely on it).

![image](https://user-images.githubusercontent.com/203839/52751341-f6e6b400-2fa3-11e9-9e2d-aa7e114455a1.png)

If I add a condition `'$(DesignTimeBuild) != ''` to the `ProjectReference` then the warning icon (and dependency) disappears but build inside VS fails.

![image](https://user-images.githubusercontent.com/203839/52751399-2c8b9d00-2fa4-11e9-8656-2b6ea506ab9c.png)

**Comment by davkean** [comment]
Haven't tried, but `Visible` on the reference should be respected. If not, that's probably a different bug.

**Comment by ReubenBond** [comment]
Visible does hide the project, but the warning icon on the Dependencies node remains

**Comment by davkean** [comment]
@drewnoakes This will be a case of dependency node expecting a counterpart design-time builds result for the item it finds in evaluation.

**Comment by drewnoakes** [comment]
Seems likely. I'll test my repro solution for this issue as part of the work I'm doing on #4550.

**Comment by jhudsoncedaron** [comment]
We track this issue. Same ultimate cause: project has a special build step which causes it to need a non-build dependency on another project. Useless visual studio warning icon with no actual warning.

**Comment by davidwengier** [comment]
The established norm for referencing source generators as a project reference is to use `OutputItemType="Analyzer" ReferenceOutputAssembly="false"` so hits for this could start to increase with .NET 5. I mention this mainly so that searching for "Source Generator" correctly finds this issue, and because I happen to be using source generators a bunch at the moment 😁

**Comment by dmonroym** [comment]
Related to https://github.com/dotnet/msbuild/issues/1916

**Comment by drewnoakes** [comment]
A fix for this issue has been merged and will be released as part of 16.8.
